app/uk/gov/hmrc/eoricommoncomponent/frontend/views/helpers/DateFormatter.scala

```diff
@@ -39,7 +40,7 @@ class DateFormatter @Inject() (languageUtils: LanguageUtils) {
     tryConvert.getOrElse(dateString)
   }
 
-  def formatLocalDate(date: LocalDate)(implicit messages: Messages): String =
-    try Some(date.format(longDateFormatter)).getOrElse(throw new Exception("Cannot convert date"))
+  def formatLocalDate(date: LocalDate): String =
+    Try(date.format(longDateFormatter)).getOrElse(throw new Exception("Cannot convert date"))
```

Comment:
For this situation what do we want to do? The formatter is only used in one view, and while it's incredible unlikely - if this throws an exception then they can't view this page or rectify the issue at all. And we're also not logging the original exception.

Personally I feel like we should have our getOrElse return "-" in this scenario such that the page will display but without the date. We know we definitely have a local date so there's no issue for the user to continue past this page, but the exception would prevent that.

If we don't want to do that I vote we don't alter the exception message so that we get a really good understanding of what went wrong in the exception message.

We'll want to add a logger.error for either of these scenarios really so that we can easily find out what went wrong in kibana (for the latter option, we can recover then log the error then rethrow the same exception).

What do we think @poscala @RobertLaverick ?

Reply:
I think adding the logger.error makes sense. Do we want to include the bad input? As much as it is rare, it would at least make it easy to reproduce if this becomes a problem in future?

Reply:
Agreed with the above - so I think we go for something like:
```
Try(date.format(longDateFormatter)) match {
  case Success(date) => date
  case Failure(e) => 
    logger.error("Cannot convert date", e)
    throw e
}
```

I'd prefer not to throw an exception, but you're right - once it's in place we can be sure it should always work without fail.